src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -1267,6 +1267,8 @@ bool fuse_type_to_constant(const std::shared_ptr<ov::Node>& node,
             new_const = change_constant_precision<ov::element::Type_t::boolean, ov::element::Type_t::u8>(constant);
         } else if (from == ov::element::boolean && to == ov::element::i32) {
             new_const = change_constant_precision<ov::element::Type_t::boolean, ov::element::Type_t::i32>(constant);
+        } else if (from == ov::element::i8 && to == ov::element::i64) {
+            new_const = change_constant_precision<ov::element::Type_t::i8, ov::element::Type_t::i64>(constant);
```

Comment:
@vurusovs how do you think we change transformation to solve issue of ngraph::pass::ConvertPrecision?

Reply:
For me it looks great that we found such limitation. @itikhono please, take a look

Reply:
this is a valid fix for this transformation.

But I'm not 100% sure, why we skipped some <from, to> pairs here.
E.g I don't see int8 to int32, uint8 to in64
Maybe the initial idea was not to have many template instantiations and add them on request.

Reply:
There are other similar issues, for example: https://github.com/openvinotoolkit/openvino/pull/21358#discussion_r1409020311

It convert i8/u8 to f32, also encounter the similar issue, should we handle it by the similar method? @itikhono

Reply:
Are these only test relating issues or do we have some real models which require such changes?

Reply:
I only found such issue in test cases, not sure whether there are any real models have such case.